<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PermissionsWidget</class>
 <widget class="QWidget" name="PermissionsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>819</width>
    <height>540</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>User informations</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <layout class="QGridLayout" name="gridLayout_3" rowstretch="0,0,0,0,0,0,0,0,0,0,1">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="2" column="0" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QRadioButton" name="rbtn_basic">
       <property name="text">
        <string>Basique</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="rbtn_hash">
       <property name="text">
        <string>Hash</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="rbtn_oauth">
       <property name="text">
        <string>OAuth2</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="10" column="0" colspan="3">
    <widget class="QTableView" name="tbv_permissions"/>
   </item>
   <item row="1" column="0" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="lbl_name">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Nom</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lne_name"/>
     </item>
    </layout>
   </item>
   <item row="9" column="0" colspan="3">
    <widget class="QLabel" name="lbl_permissions">
     <property name="text">
      <string>Sélectionner les services auxquels cette clé donne accès</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QLineEdit" name="lne_user_agent"/>
   </item>
   <item row="8" column="1">
    <widget class="QLineEdit" name="lne_referer"/>
   </item>
   <item row="8" column="0">
    <widget class="QLabel" name="lbl_referer">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Referer</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="lbl_user_agent">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>User-agent</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QStackedWidget" name="stack_key">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page_basic">
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QGroupBox" name="gpb_authentication">
         <property name="title">
          <string>Authentification</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="0">
           <widget class="QLabel" name="lbl_password">
            <property name="text">
             <string>Mot de passe</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="lbl_login">
            <property name="text">
             <string>Nom d'utilisateur</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lne_password">
            <property name="echoMode">
             <enum>QLineEdit::PasswordEchoOnEdit</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lne_login"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_hash">
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="lbl_hash">
         <property name="text">
          <string>Une valeur de hash (chaine de caractère) sera calculée automatiquement.
La valeur devra être fournie comme paramètre supplémentaire dans les requêtes de consultation des flux.
Ce type d’authentification est adapté en particulier pour une utilisation sur un site internet.</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_oauth"/>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QgsCollapsibleGroupBox" name="gpb_ip_filter">
     <property name="title">
      <string>Filtrage par IP</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="0">
       <widget class="UserKeyIpFilterWidget" name="wdg_user_key_ip_filter" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>UserKeyIpFilterWidget</class>
   <extends>QWidget</extends>
   <header>geoplateforme.gui.user_keys.wdg_user_key_ip_filter</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>lne_name</tabstop>
  <tabstop>rbtn_basic</tabstop>
  <tabstop>rbtn_hash</tabstop>
  <tabstop>rbtn_oauth</tabstop>
  <tabstop>lne_login</tabstop>
  <tabstop>lne_password</tabstop>
  <tabstop>gpb_ip_filter</tabstop>
  <tabstop>lne_user_agent</tabstop>
  <tabstop>lne_referer</tabstop>
  <tabstop>tbv_permissions</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
